xdsl/dialects/arith.py

```diff
@@ -141,29 +141,42 @@ def from_float_and_width(
         return Constant.create(result_types=[typ], attributes={"value": val})
 
 
-class BinaryOperation(IRDLOperation):
+_T = TypeVar("_T")
```

Comment:
```suggestion
_T = TypeVar("_T", bound=Attribute)
```
So people writing `BinaryOperation[int]` will get an error.